app/helpers/interactive_page_helper.rb

```diff
@@ -26,11 +26,17 @@ def page_link(activity,page, opts={})
     return link_to name, runnable_activity_page_path(activity,page), opts
   end
 
+  def main_section_hidden_plugins(page, run)
+    page.main_visible_embeddables.select { |e| e.is_hidden_plugin? }
+  end
+
   def main_section_visible_embeddables(page, run)
     finder = Embeddable::AnswerFinder.new(run)
     # Limit visible embeddables to ones that do not belong to any section.
     # Don't return answer objects for interactives.
-    page.main_visible_embeddables.map { |e| Embeddable::is_interactive?(e) ? e : finder.find_answer(e) }
+    page.main_visible_embeddables
+      .select { |e| !e.is_hidden_plugin? }
```

Comment:
it feels a bit strange `main_visible_embeddables` returns hidden plugins.
Could we just move this `.select {...}` to `main_visible_embeddables`?
So, `main_visible_embeddables` would return whatever it returns now + `.select { |e| !e.is_hidden_plugin? }`.